rustyms/src/identification/formats/msfragger.rs

```diff
@@ -0,0 +1,165 @@
+                location.full_line(),
+                location.location.clone(),
+                custom_database,
+                &SloppyParsingParameters {ignore_prefix_lowercase_n: true, ..Default::default()},
+            )});
+        protein: FastaIdentifier<String>, |location: Location, _| location.parse(IDENTIFIER_ERROR);
+        theoretical_mass: Mass, |location: Location, _| location.parse::<f64>(NUMBER_ERROR).map(Mass::new::<crate::system::dalton>);
+        delta_mass: Mass, |location: Location, _| location.parse::<f64>(NUMBER_ERROR).map(Mass::new::<crate::system::dalton>);
+        missed_cleavages: usize, |location: Location, _| location.parse(NUMBER_ERROR);
+        hyperscore: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        nextscore: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        expectscore: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        score_without_delta_mass: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        best_score_with_delta_mass: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        second_best_score_with_delta_mass: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        delta_score: f64, |location: Location, _| location.parse(NUMBER_ERROR);
+        num_matched_ions: usize, |location: Location, _| location.parse(NUMBER_ERROR);
+        tot_num_ions: usize, |location: Location, _| location.parse(NUMBER_ERROR);
+        num_tol_term: usize, |location: Location, _| location.parse(NUMBER_ERROR);
+        modification_info: Vec<(SequencePosition, SimpleModification)>, |location: Location, custom_database: Option<&CustomDatabase>| location.or_empty().array(',').map(|m| if let Some((head, tail)) = m.clone().split_once('(') {
+            let head_trim = head.as_str().trim();
+            Ok((
+            if head_trim.eq_ignore_ascii_case("N-term") {
+                SequencePosition::NTerm
+            } else if head_trim.eq_ignore_ascii_case("C-term") {
+                SequencePosition::CTerm
+            } else {
+                // Format: `14M` so take only the numeric part
+                head.as_str()[..head.len()-1].trim().parse::<usize>().map(|i| SequencePosition::Index(i-1)).map_err(|err| CustomError::error(
+                    "Invalid FragPipe modification location",
+                    format!("The location number {}", explain_number_error(&err)),
+                    head.context(),
+                ))?
+            },
+            Modification::sloppy_modification(tail.full_line(), tail.location.clone(), None, custom_database)?
+        ))
+        } else {
+            Err(CustomError::error(
+                "Invalid FragPipe modification",
+                "The format `location(modification)` could not be recognised",
+                m.context(),
+            ))
+        }).collect::<Result<Vec<_>, _>>();
+
+        // modification_info,  best_locs, localization_scores
```

Comment:
For localisation score I think copy and pasting from the FragPipe one should work, the format looks the same. Also `modification_info` is parsed so can be removed from this comment.